resend/request.py

```diff
@@ -36,12 +36,22 @@ def perform(self) -> Union[T, None]:
             requests.HTTPError: If the request fails
         """
         resp = self.make_request(url=f"{resend.api_url}{self.path}")
+
         # delete calls do not return a body
         if resp.text == "" and resp.status_code == 200:
             return None
 
+        # this is a safety net, if we get here it means the Resend API is having issues
+        # and most likely the gateway is returning htmls
+        if "application/json" not in resp.headers["content-type"]:
```

Comment:
I would suggest to use `req.headers.get("content-type", "")` to avoid any `KeyError`

Reply:
we can also take a different approach

```python
try:
     dict_response = resp.json()
except requests.exceptions.JSONDecodeError:
     raise_for_code_and_type(
                code=500,
                message="Failed to parse Resend API response. Please try again.",
                error_type="InternalServerError",
            )
```